js/src/UXCapture.js

```diff
@@ -56,8 +63,34 @@ export default class UXCapture {
     }
   }
 
+  /**
+   * Assigns user-specified `onMark` and `onMeasure` callbacks
+   * to UXCapture instance.
+   *
+   * @param {object} configuration
+   */
   config(configuration) {
-    const { onMark = NOOP, onMeasure = NOOP } = configuration;
-    Object.assign(this, { onMark, onMeasure });
+    const { onMark = NOOP, onMeasure = NOOP, interactive = false } = configuration;
+
+    if (interactive) {
+      this.startMark = INTERACTION_START_MARK;
```

Comment:
most of this is in my other comment, but it looks like we're expecting `UX.config` to be called each time there's an _Interactive View_? That doesn't feel right to me because the global `onMark` and `onMeasure` should ideally only be assigned once. I think the `this.startMark` assignment should happen in `expect` (or, in my suggestion, `initView`)

Reply:
+1

UX.config() is valid across all transitions on the page and meant to configure the whole library (for integrations like NewRelic, for example) and not individual views.

As for startTransition method, it is specific to interactive transitions and all it does, it creates a new mark on Performance Timeline (@sadafie renamed my original "transitionStart" into "interactionStart").

In page lifecycle, first view is always a page view and all subsequent views are always interactive views - see the diagram (I might need to explain it over zoom):
https://meetup.atlassian.net/wiki/spaces/WEG/pages/718700545/UX+Capture

The `initView()` method will be coming with that API redesign (WP-829) and will be handling the knowledge of page vs interactive view.

Reply:
I'll probably rename `initView()` into `startView()` though.